File: src/domain/models/detected_tool.py
"""DetectedTool value object.

Immutable representation of a detected AI coding tool, including
its name, optional configuration path, and optional version.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from pathlib import Path


@dataclass(frozen=True)
class DetectedTool:
    """An AI coding tool detected on the system.

    Attributes:
        name: Tool identifier (e.g. "claude-code", "cursor", "roo-code").
        config_path: Relative path to the tool's config directory (relative to
            home), or None. Resolution to absolute is infrastructure's job.
        version: Detected version string, or None if unknown.
    """

    name: str
    config_path: Path | None = None
    version: str | None = None

File: internal/discovery/domain/detected_tool.go
package domain

// DetectedTool is an immutable value object representing a detected AI coding tool.
type DetectedTool struct {
	name       string
	configPath string
	version    string
}

// NewDetectedTool creates a DetectedTool value object.
// Use empty string for configPath/version if unknown.
func NewDetectedTool(name, configPath, version string) DetectedTool {
	return DetectedTool{name: name, configPath: configPath, version: version}
}

// Name returns the tool identifier.
func (dt DetectedTool) Name() string { return dt.name }

// ConfigPath returns the config path (empty if unknown).
func (dt DetectedTool) ConfigPath() string { return dt.configPath }

// Version returns the version string (empty if unknown).
func (dt DetectedTool) Version() string { return dt.version }

// Equal returns true if two DetectedTools have the same values.
func (dt DetectedTool) Equal(other DetectedTool) bool {
	return dt.name == other.name && dt.configPath == other.configPath && dt.version == other.version
}
